src/server/input/cursor_controller.cpp

```diff
@@ -275,3 +281,30 @@ void mi::CursorController::cursor_moved_to(float abs_x, float abs_y)
 
     cursor->move_to(new_location);
 }
+
+void mir::input::CursorController::pointer_usable()
+{
+    bool became_usable = false;
+    std::shared_ptr<mg::CursorImage> image;
+    {
+        std::lock_guard<std::mutex> lock(cursor_state_guard);
+        became_usable = !usable;
+        image = current_cursor;
+        usable = true;
+    }
+
+    if (became_usable)
+    {
+        if (image && !is_empty(image))
+            cursor->show(*image);
```

Comment:
In theory a `pointer_unusable()` could sneak in between unlocking and calling `cursor->show(*image);`. This would leave the cursor in an incorrect state and could lead to `usable` not matching whether the cursor is actually shown. IIRC calling `hide()`/`show()` under lock is a bad idea, so I'm not sure the best way to resolve this.

Reply:
> In theory a `pointer_unusable()` could sneak in between unlocking and calling

In practice these calls would be the result of two input events serialized on the input thread. But yes, having state spread across objects like this doesn't aid consistency. I'll think about how we can organize things better.